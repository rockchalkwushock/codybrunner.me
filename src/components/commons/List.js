import styled from 'styled-components'
import { bool, number, string } from 'prop-types'

const List = styled.ul`
  display: grid;
  grid-auto-flow: ${({ flow }) => flow};
  grid-gap: ${({ gap }) => `${gap}px`};
  grid-template-columns: ${({ tags }) =>
    tags ? 'repeat(auto-fit, minmax(80px, max-content))' : '1fr'};
  padding-top: ${({ tags }) => (tags ? '0.75em' : '0')};

  @media (min-width: ${({ theme }) => theme.screen.large}) {
    grid-template-columns: ${({ tags }) =>
      tags ? 'repeat(auto-fill, minmax(140px, 1fr))' : '1fr'};
  }
`

List.defaultProps = {
  flow: 'row',
  gap: 5
}
List.displayName = 'List'
List.propTypes = {
  flow: string,
  gap: number,
  tags: bool
}

export default List
