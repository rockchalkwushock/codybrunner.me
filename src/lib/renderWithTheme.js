import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
