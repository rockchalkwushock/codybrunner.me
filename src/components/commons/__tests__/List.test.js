import React from 'react'

import renderWithTheme from '../../../lib/renderWithTheme'
import List from '../List'

describe('Component: <List />', () => {
  let forTags
  let wrapper
  beforeEach(() => {
    forTags = renderWithTheme(<List tags />)
    wrapper = renderWithTheme(<List />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('display', 'grid')
    expect(wrapper).toHaveStyleRule('grid-auto-flow', 'row')
    expect(wrapper).toHaveStyleRule('grid-gap', '5px')
    expect(wrapper).toHaveStyleRule('grid-template-columns', '1fr')
    expect(wrapper).toHaveStyleRule('padding-top', '0')
    expect(wrapper).toHaveStyleRule('grid-template-columns', '1fr', {
      media: '(min-width:736px)'
    })
  })
  test('should have styles for tags', () => {
    expect(forTags).toHaveStyleRule('display', 'grid')
    expect(forTags).toHaveStyleRule('grid-auto-flow', 'row')
    expect(forTags).toHaveStyleRule('grid-gap', '5px')
    expect(forTags).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fit,minmax(80px,max-content))'
    )
    expect(forTags).toHaveStyleRule('padding-top', '0.75em')
    expect(forTags).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fill,minmax(140px,1fr))',
      {
        media: '(min-width:736px)'
      }
    )
  })
  test('should be able to change flow and gap on props', () => {
    expect(wrapper.children().props()).toHaveProperty('flow', 'row')
    expect(wrapper.children().props()).toHaveProperty('gap', 5)
    wrapper = renderWithTheme(<List flow="column" gap={10} />)
    expect(wrapper.children().props()).toHaveProperty('flow', 'column')
    expect(wrapper.children().props()).toHaveProperty('gap', 10)
  })
})
