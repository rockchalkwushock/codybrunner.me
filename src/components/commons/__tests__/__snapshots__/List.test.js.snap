// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <List /> should render correctly 1`] = `
.c0 {
  display: grid;
  grid-auto-flow: row;
  grid-gap: 5px;
  grid-template-columns: 1fr;
  padding-top: 0;
}

@media (min-width:736px) {
  .c0 {
    grid-template-columns: 1fr;
  }
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "fg": "",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
      },
    }
  }
>
  <List
    flow="row"
    gap={5}
  >
    <ul
      className="c0"
    />
  </List>
</ThemeProvider>
`;
